frontend/degree-plan/components/FourYearPlanPage.tsx

```diff
@@ -104,6 +104,7 @@ const FourYearPlanPage = ({
   const ref = useRef(null);
 
   // search panel
+  const [searchedRuleId, setSearchedRuleId] = useState(-1);
```

Comment:
search state is in a [context](https://github.com/pennlabs/penn-courses/blob/9d83d4dcab84f1d16924e2b7b669879fbf68219d/frontend/degree-plan/components/Search/SearchPanel.tsx#L20) since we set it from a bunch of different places in the UI... do we also need this here?

Reply:
changed this to using searchRuleId which already exists instead of a new state variable